java-squid/src/main/java/org/sonar/java/cfg/CFG.java

```diff
@@ -723,54 +820,17 @@ private void buildConditionalAnd(BinaryExpressionTree conditionalAnd, Block true
   private Block createBranch(Tree terminator, Block trueBranch, Block falseBranch) {
     Block result = createBlock();
     result.terminator = terminator;
-    result.successors.add(trueBranch);
-    result.successors.add(falseBranch);
+    result.addTrueSuccessor(trueBranch);
+    result.addFalseSuccessor(falseBranch);
     return result;
   }
 
-  public void debugTo(PrintStream out) {
-    for (Block block : Lists.reverse(blocks)) {
-      if (block.id != 0) {
-        out.println("B" + block.id + ":");
-      } else {
-        out.println("B" + block.id + " (Exit) :");
-      }
-      int i = 0;
-      for (Tree tree : block.elements()) {
-        out.println("  " + i + ": " + syntaxNodeToDebugString(tree));
-        i++;
-      }
-      if (block.terminator != null) {
-        out.println("  T: " + syntaxNodeToDebugString(block.terminator));
-      }
-      if (!block.successors.isEmpty()) {
-        out.print("  Successors:");
-        for (Block successor : block.successors) {
-          out.print(" B" + successor.id);
-        }
-        out.println();
-      }
-    }
-    out.println();
-  }
-
-  private static String syntaxNodeToDebugString(Tree syntaxNode) {
-    StringBuilder sb = new StringBuilder(syntaxNode.kind().name())
-      .append(' ').append(Integer.toHexString(syntaxNode.hashCode()));
-    switch (syntaxNode.kind()) {
-      case VARIABLE:
-        sb.append(' ').append(((VariableTree) syntaxNode).simpleName().name());
-        break;
-      case IDENTIFIER:
-        sb.append(' ').append(((IdentifierTree) syntaxNode).identifierToken().text());
-        break;
-      case INT_LITERAL:
-        sb.append(' ').append(((LiteralTree) syntaxNode).token().text());
-        break;
-      default:
-        //no need to debug other syntaxNodes
+  private static void methodNodeToDebugString(MethodInvocationTree method, StringBuilder sb) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Private method 'methodNodeToDebugString' is never used. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)